spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/KubernetesConfigDataLoader.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.commons.config;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.boot.ConfigurableBootstrapContext;
+import org.springframework.boot.context.config.ConfigData;
+import org.springframework.boot.context.config.ConfigData.Option;
+import org.springframework.boot.context.config.ConfigDataLoader;
+import org.springframework.boot.context.config.ConfigDataLoaderContext;
+import org.springframework.boot.context.config.ConfigDataResourceNotFoundException;
+import org.springframework.core.Ordered;
+import org.springframework.core.env.PropertySource;
+
+/**
+ * @author Ryan Baxter
+ */
+public class KubernetesConfigDataLoader implements ConfigDataLoader<KubernetesConfigDataResource>, Ordered {
+
+	@Override
+	public ConfigData load(ConfigDataLoaderContext context, KubernetesConfigDataResource resource)
+			throws IOException, ConfigDataResourceNotFoundException {
+		List<PropertySource<?>> propertySources = new ArrayList<>();
+		ConfigurableBootstrapContext bootstrapContext = context.getBootstrapContext();
+		ConfigMapPropertySourceLocator configMapPropertySourceLocator = null;
+		SecretsPropertySourceLocator secretsPropertySourceLocator = null;
```

Comment:
I'm sorry to point this here, but I am governed by "there is nothing more permanent then a temporary solution", so I'll try to add my comment again. This is how I thought the code should look like here:

```
List<PropertySource<?>> propertySources = new ArrayList<>(2);
ConfigurableBootstrapContext bootstrapContext = context.getBootstrapContext();
Environment env = resource.getEnvironment();

if (bootstrapContext.isRegistered(ConfigMapPropertySourceLocator.class)) {
	propertySources.add(bootstrapContext.get(ConfigMapPropertySourceLocator.class).locate(env));
}

if (bootstrapContext.isRegistered(SecretsPropertySourceLocator.class)) {
	propertySources.add(bootstrapContext.get(SecretsPropertySourceLocator.class).locate(env));
}

// boot 2.4.5+
return new ConfigData(propertySources, propertySource -> {
```

IMHO, it is easier to read and shows exactly the intention.

Reply:
Yeah I completely messed this up, you are right, I merged you PR with the change thanks!